extensions/wikia/Wall/WallExternalController.class.php

```diff
@@ -277,6 +277,10 @@ public function deleteMessage() {
 			break;
 
 			case 'remove':
+				if (!$mw->canModerate($this->wg->User)) {
```

Comment:
Ugh, its terrible that WallMessage makes you do that.  In most cases I would recommend we fix the code so that clients of WallMessage don't have to manage WallMessage's internal state.  However, I don't want to expend too much effort in updating the Wall code since we're planning on replacing Wall/Forum code this year with Huddler's product.  Thanks for the explanation!